pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -34,40 +32,17 @@ const (
 
 // Go doesn't like this as a constant
 var (
-	downloaderObjTypes = []string{types.AppImgObj, types.BaseOsObj,
-		types.CertObj}
+	// Version set at build time
+	Version            = "No version specified"
+	debug              bool
+	debugOverride      bool                               // From command line arg
+	downloadGCTime     = time.Duration(600) * time.Second // Unless from GlobalConfig
```

Comment:
Should we make downloader wait for GlobalConfig to be published? That would eliminate these constants.. 
GlobalConfig provides these values..

Reply:
Possibly; I don't understand this part. These are used internally in `Run()` or other funcs locally.

If we can eliminate any part of complexity, however small, I am in favour of it. I still would prefer to merge the structural changes in, and then restructure.

This one I have attempted to simplify logic internally, or extract things to other files or structures entirely within this package. I tried to avoid extracting to an external package or change fundamental logic, since I was concerned it would be too big a change, and we always can do multiple later PRs.

Reply:
Sure.. all for doing it in phases. Am happy to get this in - this is definitely cleaner than how it is now..